pkg/storage/cacher.go

```diff
@@ -402,10 +403,20 @@ func filterFunction(key string, keyFunc func(runtime.Object) (string, error), fi
 			glog.Errorf("invalid object for filter: %v", obj)
 			return false
 		}
-		if !strings.HasPrefix(objKey, key) {
-			return false
+
+		segments := strings.Split(objKey, "/")
+		currPath := ""
+
+		match := false
+		for _, segment := range segments {
+			currPath = path.Join(currPath, segment)
+			if currPath == key {
```

Comment:
Hmm.. - but it's only for main directory. So e.g. filter(/ns1/pod1, /ns1/pod1) will return true,

I think that we should base on the fact that here objKey is computed from runtime.Object and we should probably assume what it returns. That said, we should allow for exact match.